source/expressions.tex

```diff
@@ -2737,7 +2737,7 @@
 
 template<typename T> concept C = requires {
   typename T::inner;    // required nested member name
-  typename S<T>;        // required class template specialization
+  typename S<T>;        // required valid \grammarterm{template-id}
```

Comment:
[temp.names] defines what it means for a _template-id_ to be valid. Do we mean this here? If so, we should add a cross-reference.

Further, this is presumably invalid for T==int, because `int::type` is nonsense. Can we say that, to parallel the next line?